core/string/translation.cpp

```diff
@@ -1004,124 +212,1503 @@ static _character_accent_pair _character_to_accented[] = {
+		Vector<String> supported_countries = String(locale_list[idx][2]).split(",", false);
+		for (int i = 0; i < supported_countries.size(); i++) {
+			info.supported_countries.insert(supported_countries[i]);
 		}
-		ptr++;
+		locale_info[locale_list[idx][0]] = info;
+		idx++;
 	}
 
-	return false;
-}
+	// Init locale script map.
+	locale_script_info.clear();
+	idx = 0;
+	while (locale_scripts[idx][0] != nullptr) {
+		LocaleScriptInfo info;
+		info.name = locale_scripts[idx][0];
+		info.script = locale_scripts[idx][1];
+		info.default_country = locale_scripts[idx][2];
+		Vector<String> supported_countries = String(locale_scripts[idx][3]).split(",", false);
+		for (int i = 0; i < supported_countries.size(); i++) {
+			info.supported_countries.insert(supported_countries[i]);
+		}
+		locale_script_info.push_back(info);
+		idx++;
+	}
 
-String TranslationServer::standardize_locale(const String &p_locale) {
-	// Replaces '-' with '_' for macOS Sierra-style locales
-	String univ_locale = p_locale.replace("-", "_");
+	// Init supported script list.
+	script_map.clear();
+	idx = 0;
+	while (script_list[idx][0] != nullptr) {
+		script_map[script_list[idx][1]] = script_list[idx][0];
+		idx++;
+	}
 
-	// Handles known non-ISO locale names used e.g. on Windows
-	int idx = 0;
+	// Init regional variant map.
+	varinat_map.clear();
+	idx = 0;
+	while (locale_varinats[idx][0] != nullptr) {
+		varinat_map[locale_varinats[idx][0]] = locale_varinats[idx][1];
+		idx++;
+	}
+
+	// Init locale renames.
+	locale_rename_map.clear();
+	idx = 0;
 	while (locale_renames[idx][0] != nullptr) {
-		if (locale_renames[idx][0] == univ_locale) {
-			univ_locale = locale_renames[idx][1];
-			break;
+		if (!String(locale_renames[idx][1]).is_empty()) {
+			locale_rename_map[locale_renames[idx][0]] = locale_renames[idx][1];
 		}
 		idx++;
 	}
 
-	return univ_locale;
+	// Init country names.
+	country_name_map.clear();
+	idx = 0;
+	while (country_names[idx][0] != nullptr) {
+		country_name_map[String(country_names[idx][0])] = String::utf8(country_names[idx][1]);
+		idx++;
+	}
+
+	// Init country renames.
+	country_rename_map.clear();
+	idx = 0;
+	while (country_renames[idx][0] != nullptr) {
+		if (!String(country_renames[idx][1]).is_empty()) {
+			country_rename_map[country_renames[idx][0]] = country_renames[idx][1];
+		}
+		idx++;
+	}
 }
 
-String TranslationServer::get_language_code(const String &p_locale) {
-	ERR_FAIL_COND_V_MSG(p_locale.length() < 2, p_locale, "Invalid locale '" + p_locale + "'.");
-	// Most language codes are two letters, but some are three,
-	// so we have to look for a regional code separator ('_' or '-')
-	// to extract the left part.
-	// For example we get 'nah_MX' as input and should return 'nah'.
-	int split = p_locale.find("_");
-	if (split == -1) {
-		split = p_locale.find("-");
+String TranslationServer::standardize_locale(const String &p_locale) const {
+	// Replaces '-' with '_' for macOS Sierra-style locales.
```

Comment:
Is the reference to "Sierra" still valid?

Reply:
`Sierra` is not relevant, it's used by all macOS versions.